plugins/modules/gitlab_instance_variable.py

```diff
@@ -0,0 +1,393 @@
+extends_documentation_fragment:
+  - community.general.auth_basic
+  - community.general.gitlab
+  - community.general.attributes
+
+attributes:
+  check_mode:
+    support: full
+  diff_mode:
+    support: none
+
+options:
+  state:
+    description:
+      - Create or delete instance variable.
+    default: present
+    type: str
+    choices: ["present", "absent"]
+  purge:
+    description:
+      - When set to C(true), delete all variables which are not untouched in the task.
+    default: false
+    type: bool
+  vars:
+    description:
+      - When the list element is a simple key-value pair, set masked and protected to false.
+      - When the list element is a dict with the keys I(value), I(masked) and I(protected), the user can
+        have full control about whether a value should be masked, protected or both.
+      - Support for group variables requires GitLab >= 13.0.
+      - A I(value) must be a string or a number.
+      - Field I(variable_type) must be a string with either C(env_var), which is the default, or C(file).
+      - When a value is masked, it must be in Base64 and have a length of at least 8 characters.
+        See GitLab documentation on acceptable values for a masked variable (U(https://docs.gitlab.com/ce/ci/variables/#masked-variables)).
+    default: {}
+    type: dict
+  variables:
+    description:
+      - A list of dictionaries that represents CI/CD variables.
+      - This modules works internal with this sructure, even if the older I(vars) parameter is used.
+    default: []
+    type: list
+    elements: dict
+    suboptions:
+      name:
+        description:
+          - The name of the variable.
+        type: str
+        required: true
+      value:
+        description:
+          - The variable value.
+          - Required when I(state=present).
+        type: str
+      masked:
+        description:
+          - Wether variable value is masked or not.
+        type: bool
+        default: false
+      protected:
+        description:
+          - Wether variable value is protected or not.
+        type: bool
+        default: false
+      variable_type:
+        description:
+          - Wether a variable is an environment variable (C(env_var)) or a file (C(file)).
+        type: str
+        choices: [ "env_var", "file" ]
+        default: env_var
+'''
+
+
+EXAMPLES = r'''
+- name: Set or update some CI/CD variables
+  community.general.gitlab_instance_variable:
+    api_url: https://gitlab.com
+    api_token: secret_access_token
+    purge: false
+    variables:
+      - name: ACCESS_KEY_ID
+        value: abc1312cba
+      - name: SECRET_ACCESS_KEY
+        value: 1337
+        masked: true
+        protected: true
+        variable_type: env_var
+
+- name: Delete one variable
+  community.general.gitlab_instance_variable:
+    api_url: https://gitlab.com
+    api_token: secret_access_token
+    state: absent
+    vars:
+      ACCESS_KEY_ID: acab1221cbad
+'''
+
+RETURN = r'''
+group_variable:
```

Comment:
Is `group_variable` the right name for the return values? I thought this module is about instance variables, not group variables.